tests/scripts/test_logging.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+var=0
+rm -f quotes-*.log quotes-*.err
+
+test_log()
+{
+    # Usage: test_log <log_str> <grep_str>
+    var=$((var + 1))
+    log_str="$1"
+    grep_str="$2"
+    log_file="quotes-$var.log"
+    ../../yosys -QTq -l $log_file -p "log $log_str"
+    if ! grep -qx "$grep_str" $log_file; then
+        echo "ERROR: Expected 'yosys> log $log_str' to log '$grep_str'." > "quotes-$var.err"
+        cat "quotes-$var.err"
+    fi
+}
+
+test_log "test" "test"
+test_log "test;" "test"
+test_log "test;;" "test"
```

Comment:
> Allows e.g. `log "yosys> flatten;;"`, logging the unquoted text `yosys> flatten;;`.

Doesn't this line test that the semicolons would be discarded?

Reply:
Yes, contrast with the later `test_log "\"test;;\"" "test;;"` which tests if putting them in quotes prevents removing them, i.e. `"\"test;;\""` -> `log "test;;"`.